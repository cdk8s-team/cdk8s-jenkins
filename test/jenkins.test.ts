import { ApiObject, Chart, JsonPatch, Testing } from 'cdk8s';
import { Jenkins } from '../src';
import {
  JenkinsSpecMasterBasePlugins as JenkinsBasePlugins,
} from '../src/imports/jenkins.io';

describe('a jenkins instance', () => {
  test('with defaults', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');

    // WHEN
    new Jenkins(chart, 'my-jenkins');

    // THEN
    expect(Testing.synth(chart)).toMatchSnapshot();
  });

  test('with namespace', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const namespace = 'jenkins-test-namespace';

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      namespace: namespace,
    });

    // THEN
    const manifest = Testing.synth(chart);
    expect(manifest).toMatchSnapshot();
    expect(manifest[0].metadata.namespace).toEqual(namespace);
  });

  test('with labels', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const labels = { customApp: 'my-jenkins' };

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      labels: labels,
    });

    // THEN
    const manifest = Testing.synth(chart);
    expect(manifest).toMatchSnapshot();
    expect(manifest[0].metadata.labels).toEqual(labels);
  });

  test('with disableCsrfProtection as true', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const disableCSRFProtection = true;

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      disableCsrfProtection: disableCSRFProtection,
    });

    // THEN
    const manifest = Testing.synth(chart);
    expect(manifest).toMatchSnapshot();
    expect(manifest[0].spec.master.disableCSRFProtection).toEqual(disableCSRFProtection);
  });

  test('with an update to a base plugin', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const basePlugins = [{
      name: 'configuration-as-code',
      version: '1.55',
    }];

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      basePlugins: basePlugins,
    });

    // THEN
    const manifest = Testing.synth(chart);
    expect(manifest).toMatchSnapshot();
    manifest[0].spec.master.basePlugins.forEach((plugin: JenkinsBasePlugins) => {
      if (plugin.name == basePlugins[0].name) {
        expect(plugin.version).toEqual(basePlugins[0].version);
      }
    });
  });

  test('with a user added plugin', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const plugins = [{
      name: 'simple-theme-plugin',
      version: '0.7',
    }];

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      plugins: plugins,
    });

    // THEN
    const manifest = Testing.synth(chart);
    expect(manifest).toMatchSnapshot();
    expect(manifest[0].spec.master.plugins[0].name).toEqual(plugins[0].name);
    expect(manifest[0].spec.master.plugins[0].version).toEqual(plugins[0].version);
  });

  test('with seed jobs', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    // Seed Jobs Example: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/deploying-jenkins/
    const seedJobs = [{
      id: 'jenkins-operator',
      targets: 'cicd/jobs/*.jenkins',
      description: 'Jenkins Operator repository',
      repositoryBranch: 'master',
      repositoryUrl: 'https://github.com/jenkinsci/kubernetes-operator.git',
    }];

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      seedJobs: seedJobs,
    });

    // THEN
    const manifest = Testing.synth(chart);
    expect(manifest).toMatchSnapshot();
    expect(manifest[0].spec.seedJobs).toEqual(seedJobs);
  });

  test('addBasePlugins function', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const basePlugins = [{
      name: 'workflow-api',
      version: '2.76',
    }];

    // WHEN
    const jenkins = new Jenkins(chart, 'my-jenkins');
    jenkins.addBasePlugins(basePlugins);

    // THEN
    const manifest = Testing.synth(chart);

    expect(manifest).toMatchSnapshot();
    manifest[0].spec.master.basePlugins.forEach((plugin: JenkinsBasePlugins) => {
      if (plugin.name == basePlugins[0].name) {
        expect(plugin.version).toEqual(basePlugins[0].version);
      }
    });
  });

  test('addPlugins function', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const plugins = [{
      name: 'simple-theme-plugin',
      version: '0.7',
    }];

    // WHEN
    const jenkins = new Jenkins(chart, 'my-jenkins');
    jenkins.addPlugins(plugins);

    // THEN
    const manifest = Testing.synth(chart);

    expect(manifest).toMatchSnapshot();
    expect(manifest[0].spec.master.plugins[0].name).toEqual(plugins[0].name);
    expect(manifest[0].spec.master.plugins[0].version).toEqual(plugins[0].version);
  });

  test('addSeedJobs function', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    // Seed Jobs Example: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/deploying-jenkins/
    const seedJobs = [{
      id: 'jenkins-operator',
      targets: 'cicd/jobs/*.jenkins',
      description: 'Jenkins Operator repository',
      repositoryBranch: 'master',
      repositoryUrl: 'https://github.com/jenkinsci/kubernetes-operator.git',
    }];

    // WHEN
    const jenkins = new Jenkins(chart, 'my-jenkins');
    jenkins.addSeedJobs(seedJobs);

    // THEN
    const manifest = Testing.synth(chart);

    expect(manifest).toMatchSnapshot();
    expect(manifest[0].spec.seedJobs).toEqual(seedJobs);
  });

  test('with all fields mentioned', () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');

    const namespace = 'jenkins-test-namespace';
    const labels = { customApp: 'my-jenkins' };
    const disableCSRFProtection = true;
    const basePlugins = [{
      name: 'configuration-as-code',
      version: '1.55',
    }];
    const plugins = [{
      name: 'simple-theme-plugin',
      version: '0.7',
    }];
    // Seed Jobs Example: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/deploying-jenkins/
    const seedJobs = [{
      id: 'jenkins-operator',
      targets: 'cicd/jobs/*.jenkins',
      description: 'Jenkins Operator repository',
      repositoryBranch: 'master',
      repositoryUrl: 'https://github.com/jenkinsci/kubernetes-operator.git',
    }];

    // WHEN
    new Jenkins(chart, 'my-jenkins', {
      namespace: namespace,
      labels: labels,
      disableCsrfProtection: disableCSRFProtection,
      basePlugins: basePlugins,
      plugins: plugins,
      seedJobs: seedJobs,
    });

    // THEN
    const manifest = Testing.synth(chart);

    expect(manifest).toMatchSnapshot();
    expect(manifest[0].metadata.namespace).toEqual(namespace);
    expect(manifest[0].metadata.labels).toEqual(labels);
    expect(manifest[0].spec.master.disableCSRFProtection).toEqual(disableCSRFProtection);
    manifest[0].spec.master.basePlugins.forEach((plugin: JenkinsBasePlugins) => {
      if (plugin.name == basePlugins[0].name) {
        expect(plugin.version).toEqual(basePlugins[0].version);
      }
    });
    expect(manifest[0].spec.master.plugins[0].name).toEqual(plugins[0].name);
    expect(manifest[0].spec.master.plugins[0].version).toEqual(plugins[0].version);
    expect(manifest[0].spec.seedJobs).toEqual(seedJobs);
  });

  test(('modify base plugin version using escape hatches'), () => {
    // GIVEN
    const app = Testing.app();
    const chart = new Chart(app, 'test');
    const basePlugins = [{
      name: 'workflow-job',
      version: '3.00',
    }];

    // WHEN
    const jenkins = new Jenkins(chart, 'my-jenkins');
    const jenkinsApiObject = ApiObject.of(jenkins);
    jenkinsApiObject.addJsonPatch(JsonPatch.replace('/spec/master/basePlugins/1', basePlugins[0]));

    // THEN
    const manifest = Testing.synth(chart);

    expect(manifest).toMatchSnapshot();
    manifest[0].spec.master.basePlugins.forEach((plugin: JenkinsBasePlugins) => {
      if (plugin.name == basePlugins[0].name) {
        expect(plugin.version).toEqual(basePlugins[0].version);
      }
    });
  });
});