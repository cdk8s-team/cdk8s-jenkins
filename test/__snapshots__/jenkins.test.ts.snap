// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a jenkins instance addPlugins function 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [
          Object {
            "name": "simple-theme-plugin",
            "version": "0.7",
          },
        ],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance addSeedJobs function 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [
        Object {
          "description": "Jenkins Operator repository",
          "id": "jenkins-operator",
          "repositoryBranch": "master",
          "repositoryUrl": "https://github.com/jenkinsci/kubernetes-operator.git",
          "targets": "cicd/jobs/*.jenkins",
        },
      ],
    },
  },
]
`;

exports[`a jenkins instance modify base plugin version using escape hatches 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "3.00",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with a user added plugin 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [
          Object {
            "name": "simple-theme-plugin",
            "version": "0.7",
          },
        ],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with all fields mentioned 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "customApp": "my-jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "jenkins-test-namespace",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1.55",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": true,
        "plugins": Array [
          Object {
            "name": "simple-theme-plugin",
            "version": "0.7",
          },
        ],
      },
      "seedJobs": Array [
        Object {
          "description": "Jenkins Operator repository",
          "id": "jenkins-operator",
          "repositoryBranch": "master",
          "repositoryUrl": "https://github.com/jenkinsci/kubernetes-operator.git",
          "targets": "cicd/jobs/*.jenkins",
        },
      ],
    },
  },
]
`;

exports[`a jenkins instance with an update to a base plugin 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1.55",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with defaults 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with disableCsrfProtection as true 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": true,
        "plugins": Array [],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with labels 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "customApp": "my-jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with namespace 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "jenkins-test-namespace",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [],
    },
  },
]
`;

exports[`a jenkins instance with seed jobs 1`] = `
Array [
  Object {
    "apiVersion": "jenkins.io/v1alpha2",
    "kind": "Jenkins",
    "metadata": Object {
      "labels": Object {
        "app": "jenkins",
      },
      "name": "cdk8s-jenkins-construct",
      "namespace": "default",
    },
    "spec": Object {
      "configurationAsCode": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "groovyScripts": Object {
        "configurations": Array [],
        "secret": Object {
          "name": "",
        },
      },
      "jenkinsAPISettings": Object {
        "authorizationStrategy": "createUser",
      },
      "master": Object {
        "basePlugins": Array [
          Object {
            "name": "kubernetes",
            "version": "1.31.3",
          },
          Object {
            "name": "workflow-job",
            "version": "1145.v7f2433caa07f",
          },
          Object {
            "name": "workflow-aggregator",
            "version": "2.6",
          },
          Object {
            "name": "git",
            "version": "4.10.3",
          },
          Object {
            "name": "job-dsl",
            "version": "1.78.1",
          },
          Object {
            "name": "configuration-as-code",
            "version": "1414.v878271fc496f",
          },
          Object {
            "name": "kubernetes-credentials-provider",
            "version": "0.20",
          },
        ],
        "containers": Array [
          Object {
            "image": "jenkins/jenkins:lts",
            "imagePullPolicy": "Always",
            "name": "jenkins-master",
            "resources": Object {
              "limits": Object {
                "cpu": "1500m",
                "memory": "3Gi",
              },
              "requests": Object {
                "cpu": "1",
                "memory": "500M",
              },
            },
          },
        ],
        "disableCSRFProtection": false,
        "plugins": Array [],
      },
      "seedJobs": Array [
        Object {
          "description": "Jenkins Operator repository",
          "id": "jenkins-operator",
          "repositoryBranch": "master",
          "repositoryUrl": "https://github.com/jenkinsci/kubernetes-operator.git",
          "targets": "cicd/jobs/*.jenkins",
        },
      ],
    },
  },
]
`;
